File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int weeks = n/7, days = n%7;
        int res = weeks*28 + (weeks*(weeks-1)/2)*7;
        res += days*(weeks+1) + days*(days-1)/2;
        return res;
    }
};

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks = n//7
        res = (weeks*28) + (7*((weeks*(weeks-1))//2))
        days = n%7
        res += (days*(weeks+1)) + (days*(days-1)//2)
        return res